### ISSUE TITLE
mpy-cross locations are moving

### ISSUE THREAD

**Issue by dhalbert** [aivarannamaa/pipkin#7]
We have moved `mpy-cross` builds into subdirectories in S3 and also have renamed the newer builds to some extent (haven't renamed the older builds). I didn't realize you have a list of these in https://github.com/aivarannamaa/pipkin/blob/7f33dc9768a0e0d65498446ba65bf0119ec7e9ed/data/circuitpython-mpy-cross.json. See the new structure at https://adafruit-circuit-python.s3.amazonaws.com/index.html?prefix=bin/mpy-cross/. Sorry for the pain this will cause. The long list was getting out of control for browsing purposes.

**Comment by dhalbert** [comment]
I am restoring the executables in your .json file to those pathnames as copies. Should be done in a few minutes. So this will restore functionality for pipkin for now.

**Comment by aivarannamaa** [comment]
Thanks for the notification! I never assumed these URL-s remain fixed, so there's no pain.

A question about macOS folders -- do the executables in `macos-11` work only for macOS 11 and newer? I noticed that `macos` folder doesn't contain builds for 7.3.3 and 8.0.X, but `macos-11` doesn't contain older than 8.0. Is this temporary?

**Comment by dhalbert** [comment]
Not sure what to do about the macos versions and names. GitHub Actions forced us to move forward to macOS 11 for builds, so that's why those executables are called `macos-11`. We haven't tested them on older macOS's, but I would guess they may not work.

The .mpy output actually hasn't changed during all of version 7 and version 8, so you don't really need separate executables. 8.0.x would work for all. 7.x and 8.x are the same. You definitely don't need separate versions for 7.0, 7.1, and 7.2. We only change .mpy on major version boundaries. It will be changed for version 9.x.x because of MicroPython upstream changes.

**Comment by dhalbert** [comment]
So is pipkin updated dynamically by Thonny? I thought pipkin used by Thonny internally, and you would have to make a new Thonny version. I'm just wondering how long I should leave the old paths there. Thanks.

**Comment by aivarannamaa** [comment]
> So is pipkin updated dynamically by Thonny?

pipkin is bundled with Thonny, but these URL-s are not built into pipkin. When pipkin detects that needs a version/variant of mpy-cross it hasn't cached yet, it downloads https://raw.githubusercontent.com/aivarannamaa/pipkin/master/data/circuitpython-mpy-cross.json first and looks up the URL to the required executable from there.

> I'm just wondering how long I should leave the old paths there.

I've now updated this links file in github, so all existing pipkin instances should automatically use new URL-s next time they need to download a mpy-cross. In short -- pipkin doesn't need the old paths anymore.

> You definitely don't need separate versions for 7.0, 7.1, and 7.2. We only change .mpy on major version boundaries.

The reason I included both major and minor in the key of the map, was that MicroPython does change mpy format at minor boundaries and this way I can use the same approach for both CP and MP.

> 7.x and 8.x are the same.

Great! I now changed the 8.0 mapping to give the urls of 7.3.2, just to be sure the macOS executable works on older macs as well.
